lib/core/src/rodsPath.cpp

```diff
@@ -555,6 +555,39 @@ clearRodsPath( rodsPath_t *rodsPath ) {
     return;
 }
 
+auto clearRodsPathInp( rodsPathInp_t* path ) -> void {
+  if (nullptr == path) {
+    return;
+  }
+
+  for (int i{}; i < path->numSrc; i++) {
+    clearRodsPath(&path->srcPath[i]);
+    clearRodsPath(&path->targPath[i]);
+  }
+  clearRodsPath(path->destPath);
+
+  memset(path, 0, sizeof(rodsPathInp_t));
+}
+
+auto freeRodsPathInpMembers( rodsPathInp_t* path ) -> void {
+  if (nullptr == path) {
+    return;
+  }
+
+  for (int i{}; i < path->numSrc; i++) {
+    clearRodsPath(&path->srcPath[i]);
+    clearRodsPath(&path->targPath[i]);
+  }
+  clearRodsPath(path->destPath);
+
+  free(path->srcPath);
+  path->srcPath = nullptr;
+  free(path->destPath);
+  path->destPath = nullptr;
+  free(path->targPath);
+  path->targPath = nullptr;
+}
```

Comment:
Should we `memset` the structure at the end here as is being done in `clearRodsPathInp`?

Alternatively, should `freeRodsPathInpMembers` and `clearRodsPathInp` be combined with an option to free the internal members? Perhaps the separate functions you have now are cleaner...

Reply:
I think I made the clear function first based off the existing clear functions, followed by the free function if that helps explain anything.

I think we could `memset(...)` it, or at the very least set the `numSrc` to zero in here too.

If we do memset in the free function, maybe it would be worth combining them and adding a `free_members` arg?

Reply:
Ah, it's not really *supposed* to be the same as `clearRodsPathInp`. Well, I still think it's helpful to go ahead and clear things while we're freeing. As for combining the functions, I defer. Keeping them separated is fine; combining is fine.

Reply:
`memset(...)` is now at the end of `freeRodsPathInpMembers(...)`. One function removed, so merging them is not a concern anymore.